src/test/javascript/spec/integration/code-editor/code-editor-student.integration.spec.ts

```diff
@@ -496,13 +496,11 @@ describe('CodeEditorStudentIntegration', () => {
         expect(saveFilesStub).to.have.been.calledOnceWithExactly([{ fileName: unsavedFile, fileContent: 'lorem ipsum' }]);
         expect(container.editorState).to.equal(EditorState.SAVING);
         expect(container.fileBrowser.status.editorState).to.equal(EditorState.SAVING);
-        expect(container.commitState).to.equal(CommitState.UNCOMMITTED_CHANGES);
-        expect(container.fileBrowser.status.commitState).to.equal(CommitState.UNCOMMITTED_CHANGES);
-        saveFilesSubject.next({ [unsavedFile]: null });
-
         // committing
         expect(commitStub).to.have.been.calledOnce;
         expect(container.commitState).to.equal(CommitState.COMMITTING);
+        expect(container.fileBrowser.status.commitState).to.equal(CommitState.COMMITTING);
```

Comment:
This test expected the status to be uncommited_changes as we did not wait for a response from saving the changes. After saving the changes the status is set to committing. Now we wait for save to return before we commit.